support-frontend/app/controllers/CreateSubscriptionController.scala

```diff
@@ -177,24 +179,22 @@ class CreateSubscriptionController(
   private def validate(
       request: Request[CreateSupportWorkersRequest],
       switches: Switches,
-  ): EitherT[Future, CreateSubscriptionError, Unit] = {
+  ): EitherT[Future, CreateSubscriptionError, Unit] = EitherT(for {
 
-    val paymentMethodEnabledValidation =
+    validationRulesResult <- CheckoutValidationRules.validate(request.body, paperRoundService)
```

Comment:
here we are changing the validate function from a pure function to one that needs a service.
Of course if it happens to be a national delivery, we don't know if it's valid without the help of paperround, so it's a tricky one.
I wonder whether having a separate check and validate specifically for the postcode if it's national delivery would keep the validate code cleaner?  What do you think?

Reply:
I’ve now split the future-requiring validation out and removed the Future from the `validate` call here: would you mind taking a look to see if this looks better?